src/runtime/bufferinterface.cs

```diff
@@ -0,0 +1,106 @@
+using System;
+using System.Runtime.InteropServices;
+
+namespace Python.Runtime
+{
+    /* buffer interface */
+    [StructLayout(LayoutKind.Sequential, CharSet = CharSet.Auto)]
+    internal struct Py_buffer {
+        public IntPtr buf;
+        public IntPtr obj;        /* owned reference */
+        [MarshalAs(UnmanagedType.SysInt)]
+        public IntPtr len;
+        [MarshalAs(UnmanagedType.SysInt)]
+        public IntPtr itemsize;  /* This is Py_ssize_t so it can be
+                             pointed to by strides in simple case.*/
+        [MarshalAs(UnmanagedType.Bool)]
+        public bool _readonly;
+        public int ndim;
+        [MarshalAs(UnmanagedType.LPStr)]
+        public string format;
+        public IntPtr shape;
+        public IntPtr strides;
+        public IntPtr suboffsets;
+        public IntPtr _internal;
+    }
+
+    public enum BufferOrderStyle
+    {
+        C,
+        Fortran,
+        EitherOne,
+    }
+
+    /* Flags for getting buffers */
+    public enum PyBUF
+    {
+        /// <summary>
+        /// Simple buffer without shape strides and suboffsets
+        /// </summary>
+        SIMPLE = 0,
+        /// <summary>
+        /// Controls the <see cref="PyBuffer.ReadOnly"/> field. If set, the exporter MUST provide a writable buffer or else report failure. Otherwise, the exporter MAY provide either a read-only or writable buffer, but the choice MUST be consistent for all consumers.
+        /// </summary>
+        WRITABLE = 0x0001,
+        /// <summary>
+        /// Controls the <see cref="PyBuffer.Format"/> field. If set, this field MUST be filled in correctly. Otherwise, this field MUST be NULL.
+        /// </summary>
+        FORMATS = 0x0004,
```

Comment:
This is originally named `FORMAT` and not `FORMATS` but for some reason i get a "name not cls compilant" waring if i use `FORMAT`